lib/evm_database_ops.rb

```diff
@@ -119,15 +118,19 @@ def self.restore(db_opts, connect_opts = {})
         backup_folder = action == :dump ? "db_dump" : "db_backup"
         uri = File.join(connect_opts[:uri], backup_folder, connect_opts[:remote_file_name])
       end
+    else
+      uri = db_opts[:local_file]
```

Comment:
I'm having trouble understanding how this relates to the local file default set in `connect_opts` above? Does it at all?

I guess it would be really helpful to run through what options are set in which hash for the different use cases. I always have to re-learn that every time I look at this code ...

Reply:
Yeah, this is kind of annoying here to, but it is similar to my explanation above:

https://github.com/ManageIQ/manageiq/pull/17894/files#r218127454

Basically, now everything is going through `MiqFileStorage`, so even local tasks need to use that interface to enable file splitting.  They also need to share a similar variable with the other tasks, so I kept `uri` for that case.  So this basically makes it so that `send_args` below is populated for the `file_storage.send` call, and I didn't want to muddy this further with a `uri || db_opts[:local_file]` below.

Reply:
Maybe it would be more clear if you did a `connect_opts[:uri] ||= "file://"` (or even straight `=`) here to show that you're only setting the `file://` so that we get the right subclass for the local case (rather than doing the reverse_merge above)? Or are there cases outside of `!db_opts[:local_file] == nil` where we still want the `file://` default for `connect_opts`?

Reply:
To be clear, I'm not requesting that this line change. Just wondering if we can simplify `connect_opts = connect_opts.reverse_merge(:uri => "file://")` and move it into this conditional.

Reply:
@carbonin sorry for the delay, but I think I might agree with this change.  Let me make sure it doesn't mess with anything, but I think this makes more sense then what I have.  Avoids the chance of mucking with the valid URI and only adds it when it doesn't exist.

Reply:
I will also add a comment here, since this is definitely not clear what is going on here.

Reply:
Comment added.